conda/common/logic.py

```diff
@@ -694,9 +697,9 @@ def LinearBound_(self, equation, lo, hi, preprocess, polarity):
             lo = max([lo, 0])
             hi = min([hi, total])
         if lo > hi:
-            return False
+            return FALSE
         if nterms == 0:
-            res = lo == 0
+            res = TRUE if lo == 0 else FALSE
```

Comment:
This one was a blast too. Took me quite some time to realize I missed that `bool` here..